R/FilteredData.R

```diff
@@ -315,29 +352,51 @@ FilteredData <- R6::R6Class( # nolint
     },
 
     #' @description
-    #' Adds a `TealDataset` object to this `FilteredData`
+    #' Adds a dataset to this `FilteredData`
     #'
-    #' Adds a dataset and preserve all attributes attached to this object.
     #' Technically `set_dataset` created `FilteredDataset` which keeps
     #' `dataset` for filtering purpose.
     #'
-    #' @param dataset (`TealDataset` or `TealDatasetConnector`)\cr
-    #'   the object containing data and attributes.
-    #' @return (`self`) invisibly this `FilteredTealData`
-    set_dataset = function(dataset) {
-      stopifnot(is(dataset, "TealDataset") || is(dataset, "TealDatasetConnector"))
-      dataname <- teal.data::get_dataname(dataset)
+    #' @param dataset_args (`list`)\cr
+    #'   containing the arguments except (`dataname`)
+    #'   needed by `init_filtered_dataset`
+    #' @param dataname (`string`)\cr
+    #'   the name of the `dataset` to be added to this object
+    #' @return (`self`) invisibly this `FilteredData`
```

Comment:
It is unclear from the docs what exactly goes into this list. It says needed by `init_filtered_dataset`, but it would not hurt to have it documented firstly here and then link from init_filtered_dataset to here, considering this method and its validation will be what ultimately decides what can go into it.

Also - we already pass a list of args, why not lump dataname with this list? One argument less to maintain xD

A reasonable thing to consider is to formalize this `dataset_args` as an s3 type. If it pops up anywhere else (`init_filtered_dataset`?), it can help with having validation separate from the logic here. Alternatively, drop the whole list thing and just pass the elements separately as arguments to this method? As it is now, it will be hard for users to guess what can and cannot go into this list.

Reply:
I guess this is a private class anyway, but it would be helpful for our devs as well.

Reply:
@nikolas-burkoff I agree, docs could be updated to include what are possible fields.

@kpagacz @nikolas-burkoff  ideally `datalabel`, `metadata` and `keys` (primary) should be an attribute of raw-data. But since inclusion of these attributes are not reflected anywhere in the R code (yet), we can't do this now.
`teal` doesn't do any operation on a raw-data which would discard column attributes (labels) or data attributes (datalabel, metadata). This is why collabels are not carried in the R6 classes anymore. I'm looking forward to allow us to include metadata and datalabel also.

Reply:
Just writing tests for this method. I had to go to a private `validate_dataset_args` method to check what it validates to write a proper test case.